tractdb_pyramid/views/storytelling_stravaview.py

```diff
@@ -27,6 +35,63 @@ def _get_admin(request):
 
     return admin
 
+@service_access_token.post()
+def set_access_token(request):
+    """ Exchange the provided code for an access token and store it
+    """
+
+    # Our strava code
+    code = request.matchdict['code']
+
+    # If we don't have a strava secret, we can't make the key exchange
+    if 'strava' not in request.registry.settings['secrets']:
+        request.response.status_int = 500
+        return
+
+    # Exchange the code for an access token
+    # The API will throw an unspecified error if code is invalid.
+    # Catch that rather than taking down the server.
+    access_token = ""
+
+    try:
+        client = StravaClient()
+        access_token = client.exchange_code_for_token(client_id=request.registry.settings['secrets']['strava']['strava_id'],
+            client_secret=request.registry.settings['secrets']['strava']['strava_secret'],
+            code=code)
+    except:
+        # Return appropriately
+        request.response.status_int = 500
+        return
+
+    # Our admin object
+    admin = _get_admin(request)
+
+    #Create the access token if it doesn't exist, or update the stored one
+    if not admin.exists_document('strava_access_token'):
+        # Store the access token
+        admin.create_document({'strava_access_token': access_token}, doc_id='strava_access_token')
+    else:
+        # Update the stored access token
+        try:
+            admin.update_document({'strava_access_token': access_token, '_id':'strava_access_token'})
```

Comment:
The text of that exception is poorly chosen.

But what's happening is that you're not passing the _rev, so your update is failing.

Modifying a document requires both an `_id` and a `_rev` field on the document. It check the `_rev` is the same value that's still in the database, which ensures we're not in the midst of  a race condition.